### DISCUSSION TITLE
Infinite recursive call having to do with fullscreen, focus change and getTabsterContext

### DISCUSSION THREAD

**Discussion by marlun** [microsoft/fluentui#29870] [Q&A] [answered]
We're using `@fluentui/react-components` and are having problems with an infinite recursive call having something to do with fullscreen and focus. We've got a component with a video and a toolbar. Everything works fine until we make it fullscreen and then change the focus, by clicking something like a button.

When profiling we can se the following:

<img width="770" alt="image" src="https://github.com/microsoft/fluentui/assets/19663/94cd57f1-c8c2-403b-a44b-f92648358608">

The function which seems to be called over and over is `getTabsterContext`. It then seems to call similar functions but not the same everytime:

<img width="1291" alt="image" src="https://github.com/microsoft/fluentui/assets/19663/7129429b-45ab-4671-aca4-4320fe5752bb">

But it keeps going and the tab freezes:

<img width="1715" alt="image" src="https://github.com/microsoft/fluentui/assets/19663/54a16222-3859-4d6a-a570-e2146f01a2f5">

@ling1726 It started when we moved from `9.35.0` to `9.37.3` and checking the git log we can see that there's been some changes around focus and tabster.

We would like to identify a specific commit that introduces the problem but we're having a hard time knowing how to test specific commits instead of released version tags of `@fluentui/react-components`.

**Comment by ling1726** [comment]
Is it possible to create a minimal repro on codesandbox using fluent UI components? It would be much faster for us to root cause and apply a fix

**Reply by marlun** [reply]
I've been trying and failing but will continue tomorrow (I'm in Sweden and it's getting late)

**Reply by marlun** [reply]
OK, this one is not very obvious. Took some time to actually get something which reproduced it in Codesandbox.

There's seems to be multiple things that need to be present for it to happen: 

1. One button which toggles between fullscreen
2. Two more buttons which you can click on with a tooltip where mountNode is set to `document.fullscreenElement`
3. A `<Menu>` is located in the same container

If I remove the `<Menu>` or if I remove the `mountNode` attribute it doesn't seem to happen.

Reproduce:

1. Click fullscreen button
2. Click a couple of times on the other buttons

In our app we can't reproduce it the exakt same way every time but we always do something like the above. In the codesandbox it always seem to happen when I click one of the buttons after going into fullscreen.

https://codesandbox.io/p/sandbox/wonderful-montalcini-lp5fy9

**Reply by ling1726** [reply]
Thanks for the repro, we will try it out and investigate

FYI @mshoho

**Reply by marlun** [reply]
Has there been any development here? Have you managed to understand the problem?

**Reply by mshoho** [reply]
Yes, it is infinite loop in virtual parents handling (a parent element references child element as virtual parent). We're following up with @ling1726 for the proper fix.

**Reply by ling1726** [reply, marked as answer]
Looking at the example I would actually call that an edge case we never considered - we've always assumed the Portal to be rendered out of DOM order to the reference element. I've created #29991 which should be fixed by #29990.

It's a nice edge case to catch - thx

**Reply by marlun** [reply]
Great! Will test it out when there's a new react-components release, thanks for looking at it ❤️

**Reply by marlun** [reply]
We've tested the new release and it seems to be fixed. Thanks for the help!
